src/tests/store.test.js

```diff
@@ -0,0 +1,33 @@
+import { loadState, STORE_VERSION, CURRENT_STORE_VERSION_KEY } from '../redux/reducers/store';
+
+describe('loadState', () => {
+  beforeEach(() => {
+    localStorage.clear();
+  });
+
+  it('should return undefined when state is null', () => {
+    expect(loadState()).toEqual(undefined);
+  });
+
+  describe('should return undefined when store versions don\'t match', () => {
+    it('should return undefined when store version in local storage is null', () => {
+      localStorage.setItem('state', 'state');
+      expect(loadState()).toEqual(undefined);
+    });
+
+    it('should return undefined when store versions don\'t match', () => {
+      localStorage.setItem(CURRENT_STORE_VERSION_KEY, STORE_VERSION - 1);
+      localStorage.setItem('state', 'state');
+      expect(loadState()).toEqual(undefined);
+    });
+  });
+
+  it('should return the state that was saved when store versions match', () => {
+    localStorage.setItem(CURRENT_STORE_VERSION_KEY, STORE_VERSION);
```

Comment:
Can you have another test like this on each and every line where the only difference is? localStorage.setItem(CURRENT_STORE_VERSION_KEY, STORE_VERSION -1)

Reply:
The test on line 18([here](https://github.com/openmrs/openmrs-ocl-client/blob/33ff2380ee1679767c34656ea7dd047a687added/src/tests/store.test.js#L22)) tests this same scenario. Changing that line without changing the assertion causes the test to fail, as expected.

Reply:
Make the assertion to be the opposite of this one.

Reply:
I have added a second assertion to the test on line 22([here](https://github.com/openmrs/openmrs-ocl-client/blob/ab1b18168e553376b2b06a71c1e12d796e5d1acf/src/tests/store.test.js#L22)).
I hope this suffices.

Reply:
What i did mean was an opposite of the test that you had before, but with only the assertion changing together with localStorage.setItem(CURRENT_STORE_VERSION_KEY, STORE_VERSION -1)

Reply:
Yes, Daniel. What I don't get is, is this not done here already?
```
    it('should return undefined when store versions don\'t match', () => {
      localStorage.setItem(CURRENT_STORE_VERSION_KEY, STORE_VERSION - 1);
      localStorage.setItem('state', JSON.stringify(expectedState));
      const loadedState = loadState();
      expect(loadedState).not.toEqual(expectedState);
      expect(loadedState).toEqual(undefined);
    });
```

Reply:
The original test had some data values of 1, 2

Reply:
Correct, and we moved these to a single constant, [here](https://github.com/openmrs/openmrs-ocl-client/blob/ab1b18168e553376b2b06a71c1e12d796e5d1acf/src/tests/store.test.js#L4).

Reply:
I mean the values you had in expectedState

Reply:
The constant is called `expectedState` and still has the same values. I am just reusing it.
![image](https://user-images.githubusercontent.com/12200860/57458705-779f0e80-727a-11e9-87ad-ae969b30d102.png)

Reply:
Did you just modify the existing test? Because what i did mean was for you to leave the test as it was and create a new one with only those changes in the original test.

Reply:
I did not modify the test we were referring to. What I did do was add another asertion to one of the other tests that were testing such a scenario in order to achieve what we were after, which according to this (https://github.com/openmrs/openmrs-ocl-client/pull/431#discussion_r282418513) was to test the opposite scenario. Having yet another test would be redundant.

> Make the assertion to be the opposite of this one.

Reply:
First add the test as i had advised and i will tell you what i did mean.

Reply:
Done!

Reply:
Don't you find this clearer? 😊

Reply:
Maybe, but I guess that's a point of perspective 😊.

Reply:
May be you can try to convince me. Why would you find the other better?

Reply:
Well, IMO I think it's cleaner to have a single test per scenario. Adding the second assertion to the first test makes sense to make it clearer as to what you are testing. The second test doesn't really give us anything the first one isn't!